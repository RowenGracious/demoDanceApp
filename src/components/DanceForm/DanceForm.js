import React from 'react';

// Single dance form renedered here with description
const DanceForm = ({ bio, imageSrc, name }) => {
  const imageUrl = `/images/${imageSrc}.jpg`;

  return (
    <div className="grid justify-center rounded overflow-hidden shadow-lg p-6 text-center ">
      <div className="mb-6">
        <div className="font-bold text-lg col-span-4 ">{name}</div>
      </div>
      <div className="mb-6 grid justify-center">
        <img
          className="w-24 h-24"
          src={imageUrl}
          animationDuration="1"
          applyAspectRatio="true"
        />
      </div>
      <div>{bio.substr(0, 70) + '...'}</div>
    </div>
  );
}

export default DanceForm;