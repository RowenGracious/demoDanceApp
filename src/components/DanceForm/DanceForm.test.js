import React from 'react';
import renderer from 'react-test-renderer';
import DanceForm from './DanceForm';

const imageSrc=  'electronic-dance';
const name= 'Electronic Dance';
const bio= 'This style of Dance is also known as Electronic Dance Music (EDM). This style started in Northern Europe during the 2000s and has become very popular in the next decades. This Dance is typically performed to electro house music based on different dance styles like disco, hip hop and freehand glow sticking.'

test('DanceForm component test', () => {
  const component = renderer.create(
    <DanceForm bio={bio} imageSrc={imageSrc} name={name}/>
  );
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
