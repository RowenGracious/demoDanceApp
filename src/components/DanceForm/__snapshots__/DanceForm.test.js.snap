// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DanceForm component test 1`] = `
<div
  className="grid justify-center rounded overflow-hidden shadow-lg p-6 text-center "
>
  <div
    className="mb-6"
  >
    <div
      className="font-bold text-lg col-span-4 "
    >
      Electronic Dance
    </div>
  </div>
  <div
    className="mb-6 grid justify-center"
  >
    <img
      animationDuration="1"
      applyAspectRatio="true"
      className="w-24 h-24"
      src="/images/electronic-dance.jpg"
    />
  </div>
  <div>
    This style of Dance is also known as Electronic Dance Music (EDM). Thi...
  </div>
</div>
`;
